confignode/src/main/java/org/apache/iotdb/confignode/client/async/datanode/AsyncDataNodeClientPool.java

```diff
@@ -168,60 +190,86 @@ public void sendAsyncRequestToDataNode(
         createRegionGroupsPlan.getRegionGroupMap().entrySet()) {
       for (TRegionReplicaSet regionReplicaSet : entry.getValue()) {
-        regionNum += regionReplicaSet.getDataNodeLocationsSize();
+        regionReplicaSet
+            .getDataNodeLocations()
+            .forEach(dataNodeLocation -> dataNodeLocations.add(dataNodeLocation));
       }
     }
-    Map<Integer, AbstractRetryHandler> handlerMap = new ConcurrentHashMap<>();
-    ConcurrentHashMap<Integer, TDataNodeLocation> dataNodeLocations = new ConcurrentHashMap<>();
-    Map<Integer, Object> req = new ConcurrentHashMap<>();
+    if (dataNodeLocations.isEmpty()) {
+      return;
+    }
     AtomicInteger index = new AtomicInteger();
-    CountDownLatch latch = new CountDownLatch(regionNum);
-    createRegionGroupsPlan
-        .getRegionGroupMap()
-        .forEach(
-            (storageGroup, regionReplicaSets) -> {
-              // Enumerate each RegionReplicaSet
-              regionReplicaSets.forEach(
-                  regionReplicaSet -> {
-                    // Enumerate each Region
-                    regionReplicaSet
-                        .getDataNodeLocations()
-                        .forEach(
-                            dataNodeLocation -> {
-                              handlerMap.put(
-                                  index.get(),
-                                  new CreateRegionHandler(
-                                      index.get(),
-                                      latch,
-                                      regionReplicaSet.getRegionId(),
-                                      dataNodeLocation,
-                                      dataNodeLocations));
-
-                              switch (regionReplicaSet.getRegionId().getType()) {
-                                case SchemaRegion:
-                                  req.put(
-                                      index.get(),
-                                      genCreateSchemaRegionReq(storageGroup, regionReplicaSet));
-                                  break;
-                                case DataRegion:
-                                  req.put(
-                                      index.get(),
-                                      genCreateDataRegionReq(
-                                          storageGroup,
-                                          regionReplicaSet,
-                                          ttlMap.get(storageGroup)));
-                              }
-                              dataNodeLocations.put(index.getAndIncrement(), dataNodeLocation);
-                            });
-                  });
-            });
-    sendAsyncRequestToDataNodeWithRetry(req, handlerMap, dataNodeLocations);
+    CountDownLatch countDownLatch = new CountDownLatch(dataNodeLocations.size());
+    for (int retry = 0; retry < retryNum; retry++) {
+      AbstractRetryHandler handler = null;
+      for (Map.Entry<String, List<TRegionReplicaSet>> entry :
```

Comment:
we only the `entry.getValue()`
so the foreach of `createRegionGroupsPlan.getRegionGroupMap().entrySet()` can be replaced with `createRegionGroupsPlan.getRegionGroupMap().values()`

same as below

Reply:
The above has been modified. 
But `entry.getKey()` needs to be used when creating the request here, so it has not been modified